### ISSUE TITLE
An implementation of SCI JIT

### ISSUE THREAD

**Issue by divs1210** [borkdude/sci#322]
```clojure
(defn small-clojure-jit-compiler
  "Takes a Clojure source string, and
  returns the compiled JavaScript source string."
  [clj-code]
  (google-closure-compiler/compile
    (str sci-source-compiled-to-js
         "\n sci.evalString(" clj-code ");")))
```

Because the Closure Compiler is [available in JavaScript](https://github.com/google/closure-compiler-js), this can be done **TODAY** 🤯.

But because its JavaScript source is incompatible with advanced optimizations, this single function in a library compiles into a 4 MB JavaScript file ☹️.

I was trying to compile Closure Compiler (originally written in Java) to JavaScript via [j2cl](https://github.com/google/j2cl) to produce a Closure-compliant version of Closure but have been unsuccessful so far.

So basically:
- a Small Clojure Compiler can be obtained from the Small Clojure Interpreter
- if we can obtain a Closure compliant JavaScript version of the Closure Compiler, although
- a Humongous Clojure Compiler written in a few lines of ClojureScript is sitting ominously on my machine **right now**

Thought you might find it interesting.

Thank you.

**Edit:** This was more of a thought exercise. An official bootstrapped ClojureSript in ClojureScript compiler already exists.
